lib/schema.js

```diff
@@ -151,9 +152,28 @@ class Metaschema {
       if (checked.some(v => v === null)) return null;
       return checked;
     }
+    if (decorator === 'Enum') {
+      const { values } = def;
+      const valid = Enum.from(values).has(value);
+      if (!valid) return null;
+      return value;
+    }
+    if (decorator === 'Flags') {
+      const { values } = def;
+      let flags;
+      try {
+        flags = Flags.from(...values).from(...value);
```

Comment:
I'd prefer to store it under `Class` property so that it can be used like this:
```js
const value = new this.domains.get(domainName).Class(...value);
```